app/component/summary/summary-row.cjsx

```diff
@@ -1,103 +1,64 @@
-        width: "calc(((100% - (#{data.legs.length} * #{MIN_SIZE})) * #{width}) + #{MIN_SIZE} - 2px)"
-
-      styleTime =
-        left: "calc(((100% - (#{data.legs.length} * #{MIN_SIZE})) * #{position}) + (#{i} * #{MIN_SIZE}))"
 
-      # Use either vehicle number or walking distance as text
-      if leg.transitLeg and leg.mode.toLowerCase() == 'subway'
-        text = " M"
-      else if leg.mode == 'WAIT'
-        # Use waittime in minutes
-        text = " #{Math.round(leg.duration / 60)}min"
-      else if leg.transitLeg and leg.route.length < 6
-        text = " #{leg.route}"
-      else if leg.transitLeg and leg.route.length >= 6
-        # This is somewhat dirty approach. Question is: what can we show when
-        # Leg's description is so long that it does not fit to screen?
-        # By enabling overflow: 'hidden' above we can kind of fix this, but also
-        # that option will mostly show garbage for user
-        text = ""
-      else
-        m = Math.ceil(leg.distance / 10) * 10
-        km = (leg.distance / 1000).toFixed(1)
-        text = if m < 1000 then "#{m}m" else "#{km}km"
+    noTransitLegs = true
+    for i of data.legs
+      if data.legs[i].transitLeg
+        noTransitLegs = false
 
-      legClasses =
-        "#{leg.mode.toLowerCase()}": !isFirstLeg
-        passive: @props.passive
-        start: isFirstLeg
-        end: isLastLeg
-
-      legs.push <span key={i + 'a'}
-        style={styleLine}
-        className={cx "line", leg.mode.toLowerCase()}>
-        <span key={i + 'b'} className={cx "summary-circle", legClasses}></span>
-        <RouteNumber mode={leg.mode.toLowerCase()} text={text} realtime={leg.realTime}/>
-      </span>
-
-      unless isLastLeg and not isEnoughRoomForLastLegStartTime
-        legTimes.push <DepartureTime
-          key={i + "depTime"}
-          departureTime={leg.startTime / 1000}
-          realtime={leg.realTime}
-          currentTime={currentTime}
-          style={styleTime} />
-
-      if isLastLeg
-        legTimes.push <DepartureTime
-          key="arrivalTime"
-          departureTime={leg.endTime / 1000}
-          realtime={leg.realTime}
-          currentTime={currentTime} />
+    for leg, i in data.legs
+      if leg.transitLeg or noTransitLegs
+        legs.push <RouteNumber
+                    mode={leg.mode}
+                    text={legTextUtil.getLegText(leg)}
+                    vertical={true}
+                    className={cx "line", leg.mode.toLowerCase()} />
 
     durationText = timeUtils.durationToString(duration)
 
-    classes = [
+    classes = cx [
       "itinerary-summary-row"
       "cursor-pointer"
       passive: @props.passive
     ]
 
-    <div className={cx classes} onClick={() => @props.onSelect(@props.hash)}>
+    <div className={classes} onClick={() => @props.onSelect(@props.hash)}>
+      <div className="itinerary-duration-and-distance">
+        <div className="itinerary-duration">
+          {moment.duration(duration).humanize()}
```

Comment:
We should use here pure minutes as in here:
https://projects.invisionapp.com/share/5X5RNXGGU#/screens/136398565